src/server.rs

```diff
@@ -360,6 +363,39 @@ impl Server {
         self.shutdown_handler = Some(Arc::new(function));
         debug!("Added shutdown handler {:?}", self.shutdown_handler);
     }
+
+    /// Get and set call list from memory store
+    pub fn get_calls_list(
+        self_: &PyCell<Self>,
+        limit_key: String,
+        limit_ttl: u32,
+        current_timestamp: u32,
+    ) -> Vec<u32> {
+        let server = self_.try_borrow_mut();
+        match server {
+            Ok(server) => {
+                let mut cache = server.memory_store.lock().unwrap();
+                let ttl = current_timestamp - limit_ttl;
+                if let Some(timestamps) = cache.get_mut(&limit_key) {
+                    let mut valid_timestamps: Vec<u32> = timestamps
```

Comment:
Can't this be done with 
```Rust
timestamps.retain(|timestamp| timestamp >= &ttl);
timestamps.push(current_timestamp);
```

Also won't this list always be ordered? Couldn't we just binary search for the current time stamp and drop all to the left? A linked list may be more efficient for this as well.